src/azure-cli/azure/cli/command_modules/appconfig/keyvalue.py

```diff
@@ -48,14 +49,35 @@ def import_config(cmd,
                   src_key=None,
                   src_label=None,
                   # from-appservice parameters
-                  appservice_account=None):
+                  appservice_account=None,
+                  skip_features=False):
+    src_features = []
+    dest_features = []
+    dest_kvs = []
+    need_feature_change = False
     # fetch key values from source
     if source == 'file':
         src_kvs = __read_kv_from_file(
             file_path=path, format_=format_, separator=separator, prefix_to_add=prefix, depth=depth)
+
+        if not skip_features:
+            # src_features is a list of KeyValue objects
+            src_features = __read_features_from_file(file_path=path, format_=format_)
+
     elif source == 'appconfig':
-        src_kvs = __read_kv_from_config_store(
-            cmd, name=src_name, connection_string=src_connection_string, key=src_key, label=src_label, prefix_to_add=prefix)
+        src_kvs = __read_kv_from_config_store(cmd, name=src_name, connection_string=src_connection_string,
+                                              key=src_key, label=src_label, prefix_to_add=prefix)
```

Comment:
It looks a little inefficient to fetch all items, remove feature flags and then fetch features if asked again. Instead, we can return two items like src_kvs, src_features = __read_kv_from_config_store.

Reply:
As discussed offline, this approach would not work if user specifies key filter like abc*. We need to make two separate calls to get all the keys and features.

---
In reply to: [355743324](https://github.com/Azure/azure-cli/pull/11467#discussion_r355743324) [](ancestors = 355743324)